BrainPortal/app/controllers/data_providers_controller.rb

```diff
@@ -819,6 +823,153 @@ def repair
     end
   end
 
+  # This action checks that the remote side of the DataProvider is
+  # accessible using SSH.
+  def check
```

Comment:
Most of this code should be implemented in the DP model instead, and just to top level code left in the controller side.